staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go

```diff
@@ -0,0 +1,42 @@
+package etcdmetrics
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	etcdRequestLatencies = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "etcd_request_latency_seconds",
+			Help:    "etcd request latency in seconds for each operation and object type.",
+			Buckets: prometheus.ExponentialBuckets(0.001, 5, 7),
+		},
+		[]string{"operation", "type"},
+	)
+)
+
+var registerMetrics sync.Once
+
+// Register all metrics.
+func Register() {
+	// Register the metrics.
+	registerMetrics.Do(func() {
+		prometheus.MustRegister(etcdRequestLatencies)
+	})
+}
+
+func RecordRequestLatency(verb, resource string, startTime time.Time) {
+	etcdRequestLatencies.WithLabelValues(verb, resource).Observe(sinceInSeconds(startTime))
+}
+
+func Reset() {
+	etcdRequestLatencies.Reset()
+}
+
+// sinceInSeconds gets the time since the specified start in seconds.
+func sinceInSeconds(start time.Time) float64 {
+	return time.Since(start).Seconds()
```

Comment:
is this worth it? Inline it.

Reply:
I think the compiler will inline it. Other place like apiserver use `sinceInSeconds `  too, it's not a big deal.